org.eclipse.jdt.core.tests.compiler/src/org/eclipse/jdt/core/tests/compiler/regression/XLargeTest.java

```diff
@@ -3773,7 +3785,7 @@ public void test012() {
 			"public class X {\n" +
 			"  void foo(String a, String b, String c, String d, String e) {\n" +
 			"    String s = a + (\n");
-	for (int i = 0; i < 1000; i++) {
+	for (int i = 0; i < 500; i++) {
```

Comment:
Does this test change indicate that the new compile strategy fails with shorter strings?

Reply:
Yes, but it's not all bad news. If that does happen, then that probably means the generated method size will be smaller the current error we throw in that scenario will be avoided.